git-cache

```diff
@@ -98,7 +102,7 @@ clone() {
         git_cache tag commit$SHA1 $SHA1 || true # ignore possibly already existing tag
         git clone --reference "${GIT_CACHE_DIR}" "${GIT_CACHE_DIR}" "${TARGET_PATH}" --branch commit$SHA1
     else
-        git clone --reference "${GIT_CACHE_DIR}" "${REMOTE}" "${TARGET_PATH}"
+        git clone "${REMOTE}" "${TARGET_PATH}"
```

Comment:
This changes the semantic significantly. Before, it would use the .gitcache as reference also if the exact commit would not be in the cache. Now it ignores the cache in that case.

I think there should be another test if the cache dir exists and only if it doesn't do a checkout without reference.